File: algorithms/0350. Intersection of Two Arrays II.cpp
class Solution
{
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2)
    {
        map<int, int> m;
        
        for (int num : nums1)
        {
            ++m[num];
        }
        
        vector<int> ans;
        
        for (int num : nums2)
        {
            if (m[num] > 0)
            {
                ans.push_back(num);
                --m[num];
            }
        }
        
        return ans;
    }
};

File: 0350. Intersection of Two Arrays II.cs
public class Solution 
{
    public int[] Intersect(int[] nums1, int[] nums2) 
    {
        var frequencies = new Dictionary<int, int>();
        
        foreach (var num in nums1)
        {
            if (frequencies.ContainsKey(num))
            {
                ++frequencies[num];
            }
            else
            {
                frequencies.Add(num, 1);
            }
        }
        
        var result = new List<int>();
        
        foreach (var num in nums2)
        {
            if (frequencies.ContainsKey(num) && frequencies[num] > 0)
            {
                result.Add(num);
                --frequencies[num];
            }
        }
        
        return result.ToArray();
    }
}